### PR TITLE
Added a paginator for getUserAddressesRequest

### PR DESCRIPTION
**Pull request by willemwy**
Added a paginator for getUserAddressesRequest

### CODE REVIEW

File: src/Requests/Users/GetUserAddressesRequest.php [hunk 1]
```diff
@@ -8,11 +9,18 @@ class GetUserAddressesRequest implements Command
...
+     * @var UserAddressCriteria
      */
-    public function __construct($userID)
+    private $criteria;
+
+    /**
+     * @param int                 $userID
+     * @param UserAddressCriteria $criteria
+     */
+    public function __construct($userID, UserAddressCriteria $criteria = null)
```
Comment (m-nel) [inline src/Requests/Users/GetUserAddressesRequest.php]: This is not optional

File: src/Requests/Users/Handlers/GetUserAddressesRequestHandler.php [hunk 2]
```diff
@@ -10,7 +10,12 @@ class GetUserAddressesRequestHandler extends AbstractGetRequestHandler
      */
     protected function configure($request)
     {
+        $criteria = $request->getCriteria();
         $this->setUrl('resource/users/%d/geos', $request->getUserID())
+            ->setParameters([
+                'page'          => $criteria->getPage(),
+                'pagesize'     => $criteria->getPageSize(),
+            ])
```
Comment (m-nel) [inline src/Requests/Users/Handlers/GetUserAddressesRequestHandler.php:18]: Please fix the formatting

File: src/Services/UserService.php [hunk 3]
```diff
@@ -91,13 +92,18 @@ public function updateProfile($userId, UserProfile $profile)
...
+     *
+     * @return \Ordercloud\Entities\Users\UserAddress[]
      */
-    public function getAddresses($userId)
+    public function getAddresses($userId, UserAddressCriteria $criteria = null)
     {
+        $criteria = $criteria != null ? $criteria : UserAddressCriteria::create();
+
         return $this->request(
             new GetUserAddressesRequest($userId)
```
Comment (m-nel) [inline src/Services/UserService.php]:
```php
new GetUserAddressesRequest($userId, $criteria ?: UserAddressCriteria::create())
```

Comment (m-nel) [inline src/Services/UserService.php]:
``` php
new GetUserAddressesRequest($userId, $criteria ?: UserAddressCriteria::create())
```

Comment (willemwy) [inline src/Services/UserService.php]: I do not like that format, but updated to actually pass the criteria

**Pull request merged**